tests/AuthService.IntegrationTests/DataAccess/BaseTestClass.cs

```diff
@@ -0,0 +1,83 @@
+        protected static readonly CancellationToken CancellationToken = CancellationToken.None;
+
+        protected readonly IServiceScope Scope;
+        protected readonly IRefreshTokenRepository RefreshTokenRepository;
+        protected readonly IRefreshTokenProvider RefreshTokenProvider;
+        protected readonly IDateTimeProvider DateTimeProvider;
+        protected readonly IUserRepository UserRepository;
+        protected readonly IRoleProvider RoleProvider;
+        protected readonly IUserProvider UserProvider;
+        protected readonly IUserRolesProvider UserRolesProvider;
+        protected readonly IDataContext DataContext;
+        protected readonly ITokenGenerator TokenGenerator;
+        protected readonly IJwtTokenGenerator JwtTokenGenerator;
+        protected readonly IUserService UserService;
+        protected readonly IHashProvider HashProvider;
+        protected readonly IAuthService AuthService;
+
+        public BaseTestClass(GrpcAppFactory factory)
+        {
+            Scope = factory.Services.CreateScope();
+            RefreshTokenRepository = factory.Services.GetRequiredService<IRefreshTokenRepository>();
+            RefreshTokenProvider = factory.Services.GetRequiredService<IRefreshTokenProvider>();
+            DateTimeProvider = factory.Services.GetRequiredService<IDateTimeProvider>();
+            UserRepository = factory.Services.GetRequiredService<IUserRepository>();
+            RoleProvider = factory.Services.GetRequiredService<IRoleProvider>();
+            UserProvider = factory.Services.GetRequiredService<IUserProvider>();
+            UserRolesProvider = factory.Services.GetRequiredService<IUserRolesProvider>();
+            DataContext = Scope.ServiceProvider.GetRequiredService<IDataContext>();
+            TokenGenerator = Scope.ServiceProvider.GetRequiredService<ITokenGenerator>();
+            JwtTokenGenerator = Scope.ServiceProvider.GetRequiredService<IJwtTokenGenerator>();
+            UserService = Scope.ServiceProvider.GetRequiredService<IUserService>();
+            HashProvider = Scope.ServiceProvider.GetRequiredService<IHashProvider>();
+            AuthService = Scope.ServiceProvider.GetRequiredService<IAuthService>();
+        }
+
+        protected async Task<User> ArrangeUser(bool isOpenPassword = false)
+        {
+            var roles = await RoleProvider.GetAll(CancellationToken);
+
+            var createUserModel = Builder<CreateUserModel>
+                .CreateNew()
+                .With(x => x.RoleIds = roles.Select(x => x.Id))
+                .Build();
+
+            var user = await UserProvider.GetUserByEmail(createUserModel.Email, CancellationToken) ??
+                await UserService.CreateUser(createUserModel, CancellationToken);
+
+            if (isOpenPassword)
+            {
+                user.PasswordHash = createUserModel.Password;
+            }
+            return user;
+        }
+
+        protected static readonly List<Role> SeedRoles = new()
```

Comment:
maybe we should make foreach on AuthRole. Because if we add new role into the seed we should support this test too (test will fail).